### PR TITLE
Added mergeMutation to perform Upsert Equivalent Operations

### PR DESCRIPTION
**Pull request by nanic**
@jexp Added MergeMutation to perform Upsert Equivalent Operations. Tested. This is part of #147

### CODE REVIEW

File: src/main/kotlin/org/neo4j/graphql/GraphQLSchemaBuilder.kt [hunk 1]
```diff
@@ -538,13 +538,33 @@ class GraphQLSchemaBuilder(val metaDatas: Collection<MetaData>) {
                     .dataFetcher { env ->
                         val params = mapOf<String, Any>(
                                 "id" to argumentValue(env,idProperty.fieldName),
-                                "properties" to toArguments(nonIdProperties,env))
+                                "properties" to toArguments(nonIdProperties,env).filter { it.value != null })
```
Comment (jexp) [inline src/main/kotlin/org/neo4j/graphql/GraphQLSchemaBuilder.kt]: what if you want to remove properties by setting them to null?

**Comment by jexp** [comment]
Needs more tests, remove the "println" and changed newlines and accidentally copied lines.
It also has to fail if the id-property is null as you cannot MERGE on null.

We also haven't discussed yet how to handle the MERGE case.

File: src/main/kotlin/org/neo4j/graphql/GraphQLSchemaBuilder.kt (hunk 1 above)
Comment (nanic) [reply src/main/kotlin/org/neo4j/graphql/GraphQLSchemaBuilder.kt]: @jexp should it not be a 'delete' rather than 'merge'?

**Comment by jexp** [comment]
ping @nanic Please see my comments.

**Comment by jexp** [comment]
ping @nanic

**Comment by jexp** [comment]
Added manually, thanks a lot.

**Comment by 007vasy** [comment]
Hi, Can I use this with the MakeAugmented Schema function? I would be great if I not have to write the merge for every type of connection (We are building a Knowledge graph for a law frim) @jexp ?

**Pull request closed without merging**